nixos/modules/services/development/jupyter/default.nix

```diff
@@ -0,0 +1,179 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.jupyter;
+
+
+  kernels = (pkgs.jupyterKernels.create  {
+    kernelDefinitions = if cfg.kernels != null
```

Comment:
I've basically had to move the null check here now.
I though of passing the `pkgs.jupyterKernels.defaultKernelConfig` as the default value for `jupyter.kernels`, but the it complains that it contains a reference to a derivation. (I can understand that having to build a derivation to create the doc is not the greatest).